packages/core/src/test-data/Loader.ts

```diff
@@ -67,56 +145,96 @@ export class DataLoader<T> extends Loader<T> {
 }
 
 export class JSONLoader<T> extends Loader<T> {
+	constructor(public filePath: string) {
+		super(filePath)
+	}
+
+	private async processData(path: string): Promise<T[]> {
+		const data = await this.read(path, { type: FileType.JSON })
+		if (Array.isArray(data)) return data
+		else return [data]
+	}
+
 	public async load(): Promise<void> {
-		const data = readFilePromise(this.filePath, 'utf8')
-		data.catch(err => {
-			console.error(err)
-		})
-
-		const jsonData: T[] = JSON.parse(await data)
-		if (Array.isArray(jsonData)) {
-			this.lines = jsonData
+		if (this.filePaths.length) {
+			const allLines: any[] = []
```

Comment:
I could be wrong but I think it should be like this
```suggestion
			const allLines: T[] = []
```

Reply:
Resolved